src/app/shared/pricing-definitions/pricing-definition/pricing-definition.component.ts

```diff
@@ -243,24 +321,50 @@ export class PricingDefinitionComponent implements OnInit {
   }
 
   public save(pricingDefinition: PricingDefinition) {
-    this.consistencyCheck(pricingDefinition);
-    if (this.currentPricingDefinitionID) {
-      this.updatePricingDefinition(pricingDefinition);
-    } else {
-      this.createPricingDefinition(pricingDefinition);
+    if (this.checkPricingDefinition(pricingDefinition)) {
+      if (this.currentPricingDefinitionID) {
+        this.updatePricingDefinition(pricingDefinition);
+      } else {
+        this.createPricingDefinition(pricingDefinition);
+      }
     }
   }
 
   public toggle(event) {
```

Comment:
Use specific toggle method: toggleTimeFrom()...
And type the event param according the control type.

Reply:
done